### PR TITLE
[webview_flutter] Add support for limitsNavigationsToAppBoundDomains

### PR DESCRIPTION
**Pull request by dupuchba**
`WebKitWebViewControllerCreationParams` doesn't allow the configuration of limitsNavigationsToAppBoundDomains

https://webkit.org/blog/10882/app-bound-domains/

It's very useful after ios14 if you want users to be able to call runJavaScript on a non local web-page.

This PR adds support for it

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by dupuchba** [comment]
FYI: I'll try to take care of tests tomorrow

**Comment by dupuchba** [comment]
hello @leighajarett ,
Just wanted to say that I am available if anyone has questions regarding this PR.

Thx

File: packages/webview_flutter/webview_flutter_wkwebview/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 3.4.5
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/CHANGELOG.md:1]: This adds a new feature, so it needs a minor version bump: 3.5.0.

File: packages/webview_flutter/webview_flutter_wkwebview/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+
+* Adds support for `limitsNavigationsToAppBoundDomains` in `WebKitWebViewControllerCreationParams`
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/CHANGELOG.md:3]: Preferred style for the `CHANGELOG` is to provide a short explanation of the feature and then state the method. e.g.

```markdown
* Adds support to limit navigation to pages within the app’s domain. See
  `WebKitWebViewControllerCreationParams.limitsNavigationsToAppBoundDomains`.
```

File: packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFWebViewConfigurationHostApiTests.m [hunk 3]
```diff
@@ -62,6 +62,26 @@ - (void)testSetAllowsInlineMediaPlayback {
   XCTAssertNil(error);
 }
 
+- (void)testsetLimitsNavigationsToAppBoundDomains {
+  if (@available(iOS 14, *)) {
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFWebViewConfigurationHostApiTests.m:66]: My preference is to use API_AVAILABLE(ios(14.0)) in the method declaration. See https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFUserContentControllerHostApiTests.m#L69

File: packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFWebViewConfigurationHostApiTests.m [hunk 4]
```diff
@@ -62,6 +62,26 @@ - (void)testSetAllowsInlineMediaPlayback {
   XCTAssertNil(error);
 }
 
+- (void)testsetLimitsNavigationsToAppBoundDomains {
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/example/ios/RunnerTests/FWFWebViewConfigurationHostApiTests.m:65]: nit:
```suggestion
- (void)testSetLimitsNavigationsToAppBoundDomains {
```

File: packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewConfigurationHostApi.m [hunk 5]
```diff
@@ -104,6 +104,18 @@ - (void)setAllowsInlineMediaPlaybackForConfigurationWithIdentifier:(nonnull NSNu
...
+- (void)setLimitsNavigationsToAppBoundDomainsForConfigurationWithIdentifier:
+            (nonnull NSNumber *)identifier
+                                                                  isLimited:
+                                                                      (nonnull NSNumber *)limit
+                                                                      error:(FlutterError *_Nullable
+                                                                                 *_Nonnull)error {
+  if (@available(iOS 14, *)) {
+    [[self webViewConfigurationForIdentifier:identifier]
+        setLimitsNavigationsToAppBoundDomains:limit.boolValue];
+  }
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewConfigurationHostApi.m:116]: This should set `error` for version < 14. See https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFUserContentControllerHostApi.m#L55

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart [hunk 6]
```diff
@@ -104,6 +110,12 @@ class WebKitWebViewControllerCreationParams
   /// Defaults to false.
   final bool allowsInlineMediaPlayback;
 
+  /// Whether to limit navigation to configured domains. (Only available for iOS > 14.0)
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart:113]: This should separate the first sentence: https://dart.dev/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

**Review by bparrishMines — COMMENTED** [review]
Thanks for the contribution!

I confirmed this is essentially an iOS only feature by looking through Android APIs. I didn't see an equivalent flag that can be set. It recommends just using `shouldOverrideUrlLoading` or network configs: https://developer.android.com/training/articles/security-config. Neither which requires setting a flag such as iOS.

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart [hunk 7]
```diff
@@ -678,6 +678,20 @@ class WKWebViewConfiguration extends NSObject {
     );
   }
 
+  /// Indicates whether the web view limits navigation to pages within the app’s domain.
+  ///
+  /// When navigation is limited, JS evaluation in unrestricted.
```
Comment (cyanglaz) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart:683]:
```suggestion
  /// When navigation is limited, Javascript evaluation is unrestricted.
```

**Comment by dupuchba** [comment]
hello @cyanglaz & @bparrishMines , I've taken into account your reviews!

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by dupuchba** [comment]
thanks @bparrishMines & @cyanglaz

**Pull request merged**